src/protections/WordList.ts

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright 2019, 2020 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { IProtection } from "./IProtection";
+import { Mjolnir } from "../Mjolnir";
+import { LogLevel, LogService } from "matrix-bot-sdk";
+import { logMessage } from "../LogProxy";
+import config from "../config";
+import { isTrueJoinEvent } from "../utils";
+
+export class WordList implements IProtection {
+
+    private justJoined: { [roomId: string]: { [username: string]: Date} } = {};
+    private badWords: RegExp;
+
+    constructor() {
+        // Create a mega-regex from all the tiny baby regexs
+        this.badWords = new RegExp(
+            "(" + config.protections.wordlist.words.join(")|(") + ")",
+            "i"
+        )
+    }
+
+    public get name(): string {
+        return 'WordList';
+    }
+
+    public async handleEvent(mjolnir: Mjolnir, roomId: string, event: any): Promise<any> {
+
+        const content = event['content'] || {};
+        const mbt = config.protections.wordlist.minutesBeforeTrusting;
```

Comment:
we have the luxury of large line limits: naming this something like `minBeforeTrusted` is entirely reasonable.